### PR TITLE
feat: load document action

### PR DESCRIPTION
**Pull request by Nebulis**
load document from URL:

- without key
	https://deploy-preview-48--verify-gov-sg.netlify.app/verify?q=%7B%22type%22:%22DOCUMENT%22,%22payload%22:%7B%22uri%22:%22https://gist.githubusercontent.com/Nebulis/49b53678e9b445f826125e4ac4f6d7a0/raw/0474e7b9c25dfdb95c046257483c218f27e60136/gistfile1.txt%22,%22redirect%22:%22https://verify.gov.sg/verify%22%7D%7D

- with key
	https://deploy-preview-48--verify-gov-sg.netlify.app/verify?q=%7B%22type%22:%22DOCUMENT%22,%22payload%22:%7B%22uri%22:%22https://gist.githubusercontent.com/Nebulis/49b53678e9b445f826125e4ac4f6d7a0/raw/ff6b53474fa24534f4f8c3c4aa89d17ac82a6289/encrypted-oa-issuer%22,%22key%22:%22c246183c5dacff3a90ab82024ba2361b4bd8f9ade0b443f7c2d0cc5eebe9c8ca%22,%22permittedAction%22:%5B%22STORE%22%5D,%22redirect%22:%22https://verify.gov.sg/verify%22%7D%7D

### CODE REVIEW

File: integration/load-action-plain-certificate.spec.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  await t.expect(StatusCheck.withText("Document has not been tampered").exists).ok();
+  await t.expect(StatusCheck.withText("Document has been issued").exists).ok();
+  await t.expect(StatusCheck.withText("Document’s issuer has been identified").exists).ok();
+  await validateIframeText("John Doe");
+});
+test("Load document from action should fail when url can't be parsed", async (t) => {
+  const action = {
+    type: "DOCUMENT",
+    payload: {
+      uri: `https://gist.githubusercontent.com/Nebulis/dc32c107fc5112ecf863b1dfa25995a9/raw/9aed3bfbbdd2b8/opencerts-ropsten-demo.json`,
```
Comment (simboonlong) [inline integration/load-action-plain-certificate.spec.ts:30]: i think this url is dead?

File: integration/load-action-plain-certificate.spec.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+test("Load document from action should fail when document does not exist", async (t) => {
+  const action = {
+    type: "DOCUMENT",
+    payload: {
+      uri: `https://gist.githubusercontent.com/Nebulis/fake-path/raw/fake-path/opencerts-ropsten-demo.json`,
+      redirect: "https://verify.gov.sg/verify",
+    },
+  };
+
+  // replace the first %22 after encoding by nothing so that the URL is not valid and can't be parsed
```
Comment (simboonlong) [inline integration/load-action-plain-certificate.spec.ts:48]: remove comment?

File: src/components/verify/verify-page.tsx [hunk 3]
```diff
@@ -57,10 +60,41 @@ export const VerifyPage: React.FunctionComponent = () => {
...
+          const parsedSearch = queryString.parse(location.search);
+          if (typeof parsedSearch.q !== "string") {
+            return;
+          }
+          const action = JSON.parse(window.decodeURI(parsedSearch.q));
+
+          setLoadDocumentStatus(Status.PENDING);
+          const WAIT = 2000;
+          const [wrappedDocument] = await Promise.all([retrieveDocument(action), wait(WAIT)]);
+          setLoadDocumentStatus(Status.REJECTED);
```
Comment (simboonlong) [inline src/components/verify/verify-page.tsx:84]: is this supposed to be `Status.RESOLVED`?

File: integration/load-action-plain-certificate.spec.ts (hunk 1 above)
Comment (Nebulis) [reply integration/load-action-plain-certificate.spec.ts:30]: changed

File: integration/load-action-plain-certificate.spec.ts (hunk 2 above)
Comment (Nebulis) [reply integration/load-action-plain-certificate.spec.ts:48]: removed

File: src/components/verify/verify-page.tsx (hunk 3 above)
Comment (Nebulis) [reply src/components/verify/verify-page.tsx]: yup, changed :)

**Review by simboonlong — APPROVED** [review]
lgtm

**Pull request merged**